elyra/metadata/app_utils.py

```diff
@@ -0,0 +1,372 @@
+        if option in self.argv_mappings.keys():
+            if isinstance(cli_option, Flag):  # flags set their value opposite their default
+                cli_option.value = not cli_option.default_value
+            else:  # this is a regular option, just set value
+                cli_option.set_value(self.argv_mappings.get(option))
+                if cli_option.required and not cli_option.value:
+                    self.log_and_exit("'{}' is a required parameter.".format(cli_option.option), display_help=True)
+            self._remove_argv_entry(option)
+        elif cli_option.required:
+            self.log_and_exit("'{}' is a required parameter.".format(cli_option.cli_option), display_help=True)
+        cli_option.processed = True
+
+    def process_cli_options(self, cli_options):
+        """For each Option instance in the list, process it according to the argv lists.
+           After traversal, if arguments still remain, log help and exit.
+        """
+        # Since we're down to processing options (no subcommands), scan the arguments
+        # for help entries and, if found, exit with the help message.
+        if self.has_help():
+            self.log_and_exit(display_help=True)
+
+        for option in cli_options:
+            self.process_cli_option(option)
+
+        # Check if there are still unprocessed arguments.  If so, and fail_unexpected is true,
+        # log and exit, else issue warning and continue.
+        if len(self.argv) > 0:
+            msg = "The following arguments were unexpected: {}".format(self.argv)
+            self.log_and_exit(msg, display_help=True)
+
+    def has_help(self):
+        """Checks the arguments to see if any match the help options.
+           We do this by converting two lists to sets and checking if
+           there's an intersection.
+        """
+        helps = set(['--help', '-h'])
+        args = set(self.argv_mappings.keys())
+        help_list = list(helps & args)
+        return len(help_list) > 0
+
+    def _remove_argv_entry(self, cli_option):
+        """Removes the argument entry corresponding to cli_option in both
+           self.argv and self.argv_mappings
+        """
+        # build the argv entry from the mappings since it must be located with name=value
+        if cli_option not in self.argv_mappings.keys():
+            self.log_and_exit("Can't find option '{}' in argv!".format(cli_option))
+
+        entry = cli_option
+        value = self.argv_mappings.get(cli_option)
+        if value:
+            entry = entry + '=' + value
+        self.argv.remove(entry)
+        self.argv_mappings.pop(cli_option)
+
+    def print_help(self):
+        self.print_description()
+
+    def print_description(self):
+        print(self.description)
+
+    def print_subcommands(self):
+        print()
+        print("Subcommands")
+        print("-----------")
+        print("Subcommands are launched as `elyra metadata cmd [args]`. For information on")
```

Comment:
Are these actually `elyra-metadata` ?

Reply:
Good catch.  This was done prior to realizing that supporting a space-separated command would require a 'launcher'.  Thanks for the updae.